zeebe/atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartition.java

```diff
@@ -273,10 +281,22 @@ public boolean shouldStepDown() {
   }
 
   public CompletableFuture<Void> stop() {
-    return server.stop();
+    // close the registry regardless of errors in server.stop()
+    return server.stop().whenComplete((unused, error) -> meterRegistry.close());
```

Comment:
:x: Don't we need to also clear it? I don't think it's sufficient to close it.

Reply:
Yes you are correct, clearing removes all metrics, while closing it closes all outstanding timers (and some background tasks)